Release 3.5.5 release notes

awholegunch opened the pull request:
Release notes for release 3.5.5. 

(Note: the large number of changes in the file(s) are because there were a large number of lines with trailing whitespace, which has been removed to match our style guide.)

Running as a Quick Build: https://builds.cask.co/browse/CDAP-DQB323-4

Page of interest: 
- **Release Notes:** https://builds.cask.co/artifact/CDAP-DQB323/shared/build-4/Docs-HTML/3.5.5/en/reference-manual/release-notes.html#release-3-5-5
- **Security Properties:** https://builds.cask.co/artifact/CDAP-DQB323/shared/build-4/Docs-HTML/3.5.5/en/admin-manual/appendices/cdap-site.html#security
- **Deprecated Properties:** https://builds.cask.co/artifact/CDAP-DQB323/shared/build-4/Docs-HTML/3.5.5/en/admin-manual/appendices/cdap-site.html#deprecated-properties
- **Security Docs:** _Enabling Authorization:_ https://builds.cask.co/artifact/CDAP-DQB323/shared/build-4/Docs-HTML/3.5.5/en/admin-manual/security/authorization.html#enabling-authorization

Commit:
Release notes for release 3.5.5. (Note: the large number of changes in
the file are because there were a large number of lines with trailing
whitespace, which has been removed to match our style guide.)

Commit:
Update release-notes.rst

Missed a closing back-tick.

sagarkapare commented on cdap-docs/reference-manual/source/release-notes.rst:
```diff
@@ -30,6 +30,50 @@ Cask Data Application Platform Release Notes
+  the form of ``protocol://host:port``. The property
+  ``security.auth.server.announce.address`` is now deprecated. It is only used if it is set
+  but ``security.auth.server.announce.urls`` has not been set.
+  ``security.auth.server.announce.address`` takes a single address in the form of
```
`security.auth.server.announce.address` this property should already be documented right and work as expected? no need to document it again in the release notes.

awholegunch replied:
No, this has not been documented in this detail before, nor elsewhere in the documentation. There is also a change in that the deprecated property (``security.auth.server.announce.address``) now accepts either `host:port` or just `host`.

sagarkapare commented on cdap-docs/reference-manual/source/release-notes.rst:
```diff
@@ -30,6 +30,50 @@ Cask Data Application Platform Release Notes
+- :cask-issue:`CDAP-8990` - New configurations have been added to control the YARN
+  application master container memory size, maximum heap memory size, and maximum non-heap
+  memory size: ``twill.java.heap.memory.ratio``, ``twill.yarn.am.memory.mb``, and
+  ``twill.yarn.am.reserved.memory.mb``.
```
nit: the parameters should be in order of the description
1. application master container memory size->`twill.yarn.am.memory.mb`
2. maximum heap memory size->`twill.java.heap.memory.ratio`
3. maximum non-heap memory size->`twill.yarn.am.reserved.memory.mb`

awholegunch replied:
Actually, no. I had a looong talk with Terence about this. Those do not correspond, which was my first thought. I need to add docs explaining how these work.

sagarkapare replied:
I still think its better to have it as - 

New configurations have been added to control the YARN application master container memory size, maximum heap memory size, and maximum non-heap memory size: ``twill.yarn.am.memory.mb``, ``twill.java.heap.memory.ratio``, and ``twill.yarn.am.reserved.memory.mb``.

sagarkapare commented on cdap-docs/reference-manual/source/release-notes.rst:
```diff
@@ -30,6 +30,50 @@ Cask Data Application Platform Release Notes
+  limit on the maximum number of events in-memory.
+
+- :cask-issue:`CDAP-8997` - Fixed an issue with uncaught exceptions so that they are
+  logged through the logger, allowing log collections for those exce
```
nit: `for those exceptions.`

awholegunch replied:
Not a nit. Good catch. Error in original Release Note on confluence (trailing line break that was encoded.)

Commit:
Edits cdap-default.xml (and MD5 hash), removes duplication in a security
property, and adds notes to the deprecated property list. Updates copyrights
as appropriate.

ecfm commented on cdap-common/src/main/resources/cdap-default.xml:
```diff
@@ -1667,9 +1667,10 @@
+      A comma-separated list of CDAP Authentication service announce URLs. Each URL is in
+      the form of protocol://host:port. These are the URLs that clients should use to
+      communicate with the Authentication Server. Leave empty to use the default value
+      generated by the Authentication Server.
```
Maybe also mention `If deprecated property "security.auth.server.announce.address" is set and this property is empty, "security.auth.server.announce.address" will be used`

awholegunch replied:
OK.

Commit:
Fix per comment.

Commit:
Revert changes to cdap-default.xml

Commit:
revert

Commit:
revert

sagarkapare commented on cdap-docs/reference-manual/source/release-notes.rst:
```diff
@@ -30,6 +30,52 @@ Cask Data Application Platform Release Notes
+  ``host:port`` or ``host``. Protocol and port (if not specified) will be added by the
+  Authentication Server to form the URL.
+
+  A default URL will be generated by the Authentication Serve if either property is not set.
```
nit: `Authentication Server`

awholegunch replied:
Fixed.

sagarkapare commented on cdap-docs/reference-manual/source/release-notes.rst:
```diff
@@ -30,6 +30,52 @@ Cask Data Application Platform Release Notes
+  but ``security.auth.server.announce.urls`` has not been set.
+
+  ``security.auth.server.announce.address`` now takes a single address in the form of either
+  ``host:port`` or ``host``. Protocol and port (if not specified) will be added by the
```
Please remove the last line `Protocol and port....`

Commit:
Fixes per comments.
